src/wp-includes/l10n.php

```diff
@@ -1988,3 +1988,118 @@ function wp_get_word_count_type() {
 
 	return $wp_locale->get_word_count_type();
 }
+
+/**
+ * Parses the `Accept-Language` header to get a list of locales.
+ *
+ * The locales are returned in the format WordPress expects,
+ * so "fr-CH" becomes "fr_CH" and "fr" becomes "fr_FR".
+ *
+ * @link https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language
+ *
+ * @since 6.7.0
+ * @access private
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @return string[] Locales list.
+ */
+function get_locales_from_accept_language_header() {
+	global $wpdb;
+
+	$locales = array();
+
+	if ( ! empty( $_SERVER['HTTP_ACCEPT_LANGUAGE'] ) && is_string( $_SERVER['HTTP_ACCEPT_LANGUAGE'] ) ) {
```

Comment:
To make the code easier to read I'd return early if the header isn't useful to us

```suggestion
	if ( empty( $_SERVER['HTTP_ACCEPT_LANGUAGE'] ) || ! is_string( $_SERVER['HTTP_ACCEPT_LANGUAGE'] ) ) {
		return $locales;
	}

```